ServerCore/Pages/Teams/Index.cshtml

```diff
@@ -11,77 +11,90 @@
 <h2>All Teams</h2>
 
 <p>
-    @if (Model.Event.IsTeamRegistrationActive && Model.EventRole == ModelBases.EventRole.admin)
+    @if (Model.EventRole == ModelBases.EventRole.admin)
     {
+        if (!Model.Event.IsTeamRegistrationActive)
+        {
+            <div class="alert alert-danger" role="alert">
+                This event is not currently open for registration.
+            </div>
+        }
+        else if (Model.Event.MaxNumberOfTeams <= Model.Teams.Count)
+        {
+            <div class="alert alert-danger" role="alert">
+                This event is full.
+            </div>
+        }
+
         <a asp-page="Create">Create New</a>
     }
-    else
-    {
-        <div class="alert alert-danger" role="alert">
-            This event is not currently open for registration.
-        </div>
-    }
 </p>
+<h4>Registered Teams: @Model.Teams.Count/@Model.Event.MaxNumberOfTeams, Registered Players: @Model.PlayerCount/@(Model.Event.MaxNumberOfTeams * Model.Event.MaxTeamSize)</h4>
 <table class="table">
     <thead>
         <tr>
             <th>
-                @Html.DisplayNameFor(model => model.Teams[0].Name)
+                @Html.DisplayNameFor(model => model.Teams.FirstOrDefault().Key.Name)
+            </th>
+            <th>
+                Size
             </th>
             <th>
-                @Html.DisplayNameFor(model => model.Teams[0].RoomID)
+                @Html.DisplayNameFor(model => model.Teams.FirstOrDefault().Key.RoomID)
             </th>
             <th>
-                @Html.DisplayNameFor(model => model.Teams[0].CustomRoom)
+                @Html.DisplayNameFor(model => model.Teams.FirstOrDefault().Key.CustomRoom)
```

Comment:
Should we add an issue to go through and fix this elsewhere? I'm pretty sure this pattern is in a few other places as well

Reply:
The issue here is that Teams stopped being an IList, so the previous code actually broke. This wasn't me being pedantic.

Reply:
To be clearer: I don't think there was anything wrong with the previous code so I don't think we need to make a pass for this.